src/libraries/shims/fakes/Directory.Build.targets

```diff
@@ -0,0 +1,30 @@
+<Project>
+
+  <Import Project="..\Directory.Build.targets" />
+
+  <PropertyGroup>
+    <TypeForwardedTypesCompilePath>$(IntermediateOutputPath)$(MSBuildProjectName).TypeForwardedTypes.cs</TypeForwardedTypesCompilePath>
+  </PropertyGroup>
+
+  <Target Name="CreateCompileSourceForTypeForwardedTypes"
+          BeforeTargets="CoreCompile"
+          Inputs="$(MSBuildProjectFullPath)"
+          Outputs="$(TypeForwardedTypesCompilePath)">
+    <Error Text="The 'TypeForwardedType' item must not be empty." Condition="'@(TypeForwardedType)' == ''" />
+
+    <ItemGroup>
+      <TypeForwardedType Namespace="$([System.String]::new('%(Identity)').Substring(0, $([System.String]::new('%(Identity)').LastIndexOf('.'))))" />
+      <TypeForwardedType TypeName="$([System.String]::new('%(Identity)').Substring($([MSBuild]::Add($([System.String]::new('%(Identity)').LastIndexOf('.')), 1))))" />
+      <TypeForwardedTypesLine Include="namespace %(TypeForwardedType.Namespace) { public class %(TypeForwardedType.TypeName) { } }" />
```

Comment:
This might not work if you have any generic types.  Also, double check if there are nested types and if their references change in any way -- perhaps you can see that by disassembling and diffing the shims produced.

Reply:
There was only a single generic type which worked by adding `T` to the type name's diamond. Will diff the produced assemblies.